### PR TITLE
feat: adding dataproc instantiate inline workflows sample

### PR DESCRIPTION
**Pull request by bradmiro**
Adding a sample showing how to instantiate an inline workflow template with the client library.

### CODE REVIEW

File: dataproc/instantiate_inline_workflow_template.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dataproc shows how you can use the Cloud Dataproc Client library to manage
+// Cloud Dataproc clusters. In this example, we'll show how to instantiate an inline
+// workflow template.
```
Comment (tbpg) [inline dataproc/instantiate_inline_workflow_template.go]: Only need the package comment in one file. So, you should be able to delete this.

File: dataproc/instantiate_inline_workflow_template_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	buf := new(bytes.Buffer)
+
+	if err := instantiateInlineWorkflowTemplate(buf, tc.ProjectID, region); err != nil {
+		t.Fatalf("instantiateInlineWorkflowTemplate got err: %v", err)
+	}
+
+	got := buf.String()
+	if want := fmt.Sprintf("successfully"); !strings.Contains(got, want) {
+		t.Fatalf("InstantiateInlineWorkflowTemplate: got %s, want %s", got, want)
```
Comment (tbpg) [inline dataproc/instantiate_inline_workflow_template_test.go]: Super minor:

```suggestion
		t.Fatalf("instantiateInlineWorkflowTemplate got %q, want %q", got, want)
```

File: dataproc/instantiate_inline_workflow_template.go (hunk 1 above)
Comment (bradmiro) [reply dataproc/instantiate_inline_workflow_template.go]: If one is looking to learn more about the package, how does one differentiate if they should check this sample or the other one? Should these all be separate packages?

Comment (tbpg) [reply dataproc/instantiate_inline_workflow_template.go]: https://pkg.go.dev/github.com/GoogleCloudPlatform/golang-samples/dataproc?tab=doc

If there are many files (like for samples), it's common to have a `doc.go` file that has this comment, the `package` line, and nothing else.

**Pull request merged**